CMakeLists.txt

```diff
@@ -8,15 +8,14 @@ cmake_minimum_required(VERSION 3.15)
 cmake_policy(SET CMP0079 NEW)
 cmake_policy(SET CMP0091 NEW)
 
-set(version "1.4.0" CACHE STRING "Project version")
-
-string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]" version_short ${version})
+set(BUILD_NUMBER "0" CACHE STRING "Build number")
+set(VERSION_STRING "1.4.0" CACHE STRING "Project version (string)")
 
 # Get compilation year
 string(TIMESTAMP MUMBLE_BUILD_YEAR "%Y")
 
 project(Mumble
-	VERSION ${version_short}
+	VERSION "1.4.0.${BUILD_NUMBER}"
```

Comment:
In cases where `BUILD_NUMBER` is not defined, we get an extraneous "." at the end of our version :point_up:

Reply:
It should always be defined though, as the default value is `0`.

Reply:
> It should always be defined though, as the default value is `0`.

This didn't seem to be the case as discussed in the meeting. I would approve of that change, though. If I need to make adjustments in my PR or drop the change entirely, I can do that. I am creating the var when `packaging=ON` since there isn't really a reason to assign it otherwise.

Reply:
> I have just seen most practices with CMake where values and options that are modified from the CLI are typically all caps.

So, like this `BUILD_NUMBER`. But I think that options should probably be in this syntax and variables in lower case with `-` or `_`. It would make identifying the types easier.

Reply:
It actually does make a difference, because this PR also changes the RC templates so that the build number is shown in the binaries' (.exe/.dll) properties.

Reply:
Okay in that case I'd say always default to 0 but in case `packaging=ON`, error if the variable is not set before default-initializing it. Like so:
```
if (packaging AND BUILD_NUMBER STREQUAL "")
    message(FATAL_ERROR "...")
endif()

set(BUILD_NUMBER "0" CACHE STRING "...")
```

Reply:
You mean that we should begin counting at `1` instead of `0`?

Reply:
No. But we can check whether the build number was explicitly set and enforce this to be the case if `packaging=ON` in order to avoid accidentally forgetting to set it when packaging Mumble

Reply:
Right, I agree.